Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -465,32 +476,32 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     }
     if let unexpected = node.unexpectedBetweenConditionAndTrailingComma,
       let availability = node.condition.as(AvailabilityConditionSyntax.self),
-      let (comparisonOperator, falseKeyword) = unexpected.twoTekens(
+      let (_, falseKeyword) = unexpected.twoTokens(
         firstSatisfying: { $0.tokenKind == .binaryOperator("==") },
         secondSatisfying: { $0.tokenKind == .keyword(.false) }
       )
     {
       // Diagnose #available used as an expression
-      let availabilityKeyword = availability.availabilityKeyword
-      let negatedAvailabilityKeyword = availabilityKeyword.negatedAvailabilityKeyword
-      addDiagnostic(
-        unexpected,
-        AvailabilityConditionAsExpression(availabilityToken: availabilityKeyword, negatedAvailabilityToken: negatedAvailabilityKeyword),
-        fixIts: [
-          FixIt(
-            message: ReplaceTokensFixIt(replaceTokens: [availabilityKeyword], replacement: negatedAvailabilityKeyword),
-            changes: [
-              .replace(oldNode: Syntax(availabilityKeyword), newNode: Syntax(negatedAvailabilityKeyword)),
-              .replaceTrailingTrivia(token: availability.rightParen, newTrivia: .zero),
-            ]
-          ),
-          FixIt(
-            message: RemoveNodesFixIt([comparisonOperator, falseKeyword]),
-            changes: .makeMissing([comparisonOperator, falseKeyword])
-          ),
-        ],
-        handledNodes: [unexpected.id]
+      let negatedAvailabilityKeyword = availability.availabilityKeyword.negatedAvailabilityKeyword
+      let negatedCoditionElement = ConditionElementSyntax(
+        condition: .availability(availability.with(\.availabilityKeyword, negatedAvailabilityKeyword)),
+        trailingComma: node.trailingComma
       )
+      if let negatedAvailability = negatedCoditionElement.condition.as(AvailabilityConditionSyntax.self) {
+        addDiagnostic(
+          unexpected,
+          AvailabilityConditionAsExpression(availabilityToken: availability.availabilityKeyword, negatedAvailabilityToken: negatedAvailabilityKeyword),
+          fixIts: [
+            FixIt(
+              message: ReplaceWithMultipleTokensFixIt(replaceTokens: getTokenRange(availability.availabilityKeyword, falseKeyword), replacements: getTokenRange(negatedAvailability.availabilityKeyword, negatedAvailability.rightParen)),
```

Comment:
Why do you need to do `getTokenRange` here? Can’t you just do `[availability.availabilityKeyword, comparisonOperator, falseKeyword]`?

Reply:
I've already tried what you suggested and in this case the diagnostic seems to be
`replace '#available== false' with '#unavailable(*)'`
instead of
`replace '#available(*) == false' with '#unavailable(*)'`

Reply:
Oh, I see. I forgot about the `(*)`. In that case is a very good solution. I’ve got a few minor comments on it:
- It’s not really returning a range, so I would name it `getTokens(between: TokenSyntax, and: TokenSyntax)`
- I would make it a top-level function instead of a member of `ParseDiagnosticGenerator`. Also I would mark it as `fileprivate` so we explicitly introduce new users of it
- You can also use this function in `visit(_ node: ForInStmtSyntax)` and thereby remove the `FIXME` in there.

Reply:
Thank you, indeed that `FIXME` gave me the idea, I was already looking at it, that would have been my next question :)